rest/route/host.go

```diff
@@ -577,3 +578,54 @@ func (rh *hostProvisioningOptionsGetHandler) Run(ctx context.Context) gimlet.Res
 	}
 	return gimlet.NewJSONResponse(apiOpts)
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// GET /rest/v2/hosts/ip_address/{ip_address}
+
+type hostIpAddressGetHandler struct {
+	IP   string
+	Host *host.Host
+	sc   data.Connector
+}
+
+func makeGetHostByIpAddress(sc data.Connector) gimlet.RouteHandler {
+	return &hostIpAddressGetHandler{
+		sc: sc,
+	}
+}
+
+func (h *hostIpAddressGetHandler) Factory() gimlet.RouteHandler {
+	return &hostIpAddressGetHandler{
+		sc: h.sc,
+	}
+}
+func (h *hostIpAddressGetHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.IP = gimlet.GetVars(r)["ip_address"]
+
+	if h.IP == "" {
+		return errors.New("missing ip_address to query by")
```

Comment:
this is better. I think it might be even better phrased as "ip_address must be specified" or "ip_address cannot be empty" so that it is clearer to the user that they can fix the error by fixing the input. It might be a good idea to phrase it like that. 
It might also be a good idea to wrap it in gimlet.ErrorResponse with a http.StatusBadRequest status code.

Reply:
done